### ISSUE TITLE
feat: Avoid hitting rate limits by looking at `X-Rate-Limit-*` headers

### ISSUE THREAD

**Issue by erezrokah** [okta/okta-sdk-golang#434]
### Describe the feature request?

Implement the best practices from https://developer.okta.com/docs/reference/rl-best-practices/#example-rate-limit-header-with-org-wide-rate-limit to avoid hitting rate limits, and also wait until the limit reset time instead of a retry/backoff mechanism.

### New or Affected Resource(s)

All existing and new resources will be affected I think

### Provide a documentation link

https://developer.okta.com/docs/reference/rl-best-practices/#example-rate-limit-header-with-org-wide-rate-limit

### Additional Information?

I'm aware of the existing backoff and retry configuration, it seems better practice to follow the headers returned from the server

**Comment by tjpoe** [comment]
agreed. we are hitting rate warnings on our instance and would love a way to throttle the speed of requests to stay within the api limitation.

**Comment by erezrokah** [comment]
Not stale

**Comment by erezrokah** [comment]
Not stale

**Comment by erezrokah** [comment]
Not stale

**Comment by erezrokah** [comment]
Proposal at https://github.com/okta/okta-sdk-golang/pull/450, will wait for feedback before adding docs, tests and updating the template files that generate the code

**Comment by erezrokah** [comment]
Not stale

**Comment by erezrokah** [comment]
Not stale

**Comment by erezrokah** [comment]
Not stale

**Comment by erezrokah** [comment]
Not stale

**Comment by erezrokah** [comment]
Not stale

**Comment by erezrokah** [comment]
Not stale

**Comment by erezrokah** [comment]
Not stale, will try to respond to https://github.com/okta/okta-sdk-golang/pull/450 soon

**Comment by erezrokah** [comment]
Not stale

**Comment by erezrokah** [comment]
Not stale

**Comment by erezrokah** [comment]
Thank you for the bump dear stale bot, not stale

**Comment by erezrokah** [comment]
Not stale

**Comment by erezrokah** [comment]
Not stale

**Comment by erezrokah** [comment]
Not stale

**Comment by erezrokah** [comment]
Not stale

**Comment by erezrokah** [comment]
Not stale

**Comment by erezrokah** [comment]
Not stale

**Comment by erezrokah** [comment]
Not stale

**Comment by erezrokah** [comment]
not stale
